app/src/main/java/com/example/soundcloud/discover/DiscoverFragment.java

```diff
@@ -1,14 +1,21 @@
 package com.example.soundcloud.discover;
 
 import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
 import android.support.v4.app.Fragment;
+import android.util.Log;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
+import android.widget.Toast;
 
 import com.example.soundcloud.R;
+import com.example.soundcloud.data.source.SongRepository;
+import com.example.soundcloud.data.source.remote.SongRemoteDataSource;
 
 public class DiscoverFragment extends Fragment {
+    private DiscoverPresenter mPresenter;
```

Comment:
```suggestion
    private DiscoverContract.Presenter mPresenter;
```
Anh nên khai báo Contract thay vì khai báo Presenter
thì contract a define mới mang lại giá trị